avocado/utils/pci.py

```diff
@@ -0,0 +1,291 @@
+    functions = process.system_output(cmd, shell=True, ignore_status=True)
+    if functions:
+        return functions.splitlines()
+
+
+def get_slot_from_sysfs(device):
+    """
+    Gets the pci slot of given device.
+
+    :parm device: pci device.
+
+    :return: slot of pci device from sysfs.
+    """
+    if not os.path.isdir('/sys/bus/pci/devices/%s' % device):
+        return
+    cmd = "cat /sys/bus/pci/devices/%s/devspec ; echo" % device
+    devspec = process.system_output(cmd, shell=True).strip('\n')
+    cmd = "cat /proc/device-tree/%s/ibm,loc-code ; echo" % devspec
+    slot = process.system_output(cmd, shell=True).strip('\0\n')
+    return re.match(r'((\w+)[\.])+(\w+)-(\w*\d+)-(\w*\d+)|Slot(\d+)',
+                    slot).group()
+
+
+def get_pci_id_from_sysfs(device):
+    """
+    Gets the pci id from sysfs of given device.
+
+    :parm device: pci device.
+
+    :return: pci id of a pci device from sysfs.
+    """
+    if not os.path.isdir('/sys/bus/pci/devices/%s' % device):
+        return
+    pci_id = ""
+    for params in ['vendor', 'device', 'subsystem_vendor', 'subsystem_device']:
+        cmd = "cat /sys/bus/pci/devices/%s/%s" % (device, params)
+        p_id = process.system_output(cmd, shell=True).strip('\n').strip('0x')
+        while len(p_id) < 4:
+            p_id = "0%s" % p_id
+        pci_id += "%s:" % p_id
+    if pci_id:
+        return pci_id[:-1]
+
+
+def get_slot_list():
+    """
+    Gets list of pci slots in the system.
+
+    :return: list of slots in the system.
+    """
+    slot_list = []
+    for device in get_pci_devices():
+        slot_list.append(get_slot_from_sysfs(device))
+    return list(set(slot_list))
+
+
+def get_specific_pci_id(device, part):
+    """
+    Gets specific pci id of given divice.
+
+    :parm device: pci device.
+    :parm part: part of pci id.
+
+    :return: specific pci id of a pci device.
+    """
+    cmd = "lspci -Dnvmm -s %s | grep -iw %s" % (device, part)
+    output = process.system_output(cmd, shell=True, ignore_status=True)
+    if not output:
+        return
+    return output.split()[-1]
+
+
+def get_pci_id(device):
+    """
+    Gets pci id of given divice.
+
+    :parm device: pci device.
+
+    :return: pci id of a pci device.
+    """
+    pci_id = ""
+    for params in ['Vendor', 'Device', 'SVendor', 'SDevice']:
+        output = get_specific_pci_id(device, params)
+        if output:
+            pci_id += "%s:" % output
+    if pci_id:
+        return pci_id[:-1]
+
+
+def get_memory_address(device):
+    """
+    Gets the memory address of a pci device.
+
+    :parm device: pci device.
+
+    :return: memory address of a pci device.
+    """
+    cmd = "lspci -bv -s %s" % device
+    output = process.system_output(cmd, shell=True, ignore_status=True)
+    if not output:
+        return
+    for line in output.splitlines():
+        if 'Memory at' in line:
+            return "0x%s" % line.split()[2]
```

Comment:
@ldoktor Yes, there can be many `Memory at` records, but the first record is what I plan to use in some of my tests. So, I returned only that.
I will include this in the readme.